src/modules/local_position_estimator/sensors/lidar.cpp

```diff
@@ -89,7 +90,7 @@ void BlockLocalPositionEstimator::lidarCorrect()
 	Matrix<float, n_y_lidar, n_y_lidar> S_I = inv<float, n_y_lidar>((C * _P * C.transpose()) + R);
 	Vector<float, n_y_lidar> r = y - C * _x;
 	_pub_innov.get().hagl_innov = r(0);
-	_pub_innov.get().hagl_innov_var = R(0, 0);
+	_pub_innov.get().hagl_innov_var = 1 / S_I(0, 0);
```

Comment:
can you explain this change?

Reply:
I had made the change to `1 / S_I` because that's technically the proper value to write into the innovation variance (and the more interesting value). I have changed it back since it's an unrelated change to this PR.

Reply:
Wait. Analysing the Kalman Filter itself, I think you are probably right here:
1. The pre-fit residual (innovation) is `r = y - C * _x`, where `r` is the innovation, y is the measurement at that time, C is the measurement matrix and `_x` is the state vector;
2. The innovation covariance is `S = R + C * _P * C.transpose()`, where `R` is the covariance of the observation noise and the `_P`is the state covariance matrix.
According to the `ek2_innovations` uORB msg, all `_innov_var` fields are the innovations variances/covariances. So I assume that we want `S` (the residual covariance) and not `R` (the noise covariance).
@jgoppert can we get your input here please?

Reply:
> I had made the change to 1 / S_I because that's technically the proper value to write into the innovation variance (and the more interesting value). I have changed it back since it's an unrelated change to this PR.

I think this needs a fix. I will PR it and ping @jgoppert on the process. Thanks for heads up @ndepal